tools/scuffed-scraper/main.py

```diff
@@ -0,0 +1,612 @@
+# DTU Scuffed Course Scraper
+# Copyright (C) 2023 Sofus Albert Høgsbro Rose
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import sys
+
+SUPPORTED_PY_MAJOR = 3
+SUPPORTED_PY3_VERSIONS = [11, 12, 13, 14]
+if not all(
+	[
+		sys.version_info.major == SUPPORTED_PY_MAJOR,
+		sys.version_info.minor in SUPPORTED_PY3_VERSIONS,
+	],
+):
+	print("Python 3.X minor version must be in:", SUPPORTED_PY3_VERSIONS)
+	sys.exit(1)
```

Comment:
Are these really strict version requirements? 12, 13 and 14 haven't been released. So I think we are guessing here.

If anything, you can choose a minimal version of Python that is supported. It would be great to see if this still works with 3.9 for instance, if so, then just have a lower requirement, no highest.
This is internal use, so we don't need the complexity for that.

Reply:
I see, so just `> 11`?

The script uses `match` syntax, so `3.11` really is a hard-minimum requirement. Knowing how Python deprecation can be, I was looking to force whomever ends up using `3.15` one day with this to have another look at this script and make sure all is well.

Reply:
Ok, then just put 11 as the min req, and thats it! :+1: